specifyweb/frontend/js_src/lib/components/FormPlugins/HostTaxon.tsx

```diff
@@ -44,26 +44,29 @@ export function HostTaxon({
               .maybe(records[0], deserializeResource)
               ?.rgetPromise('rightSideCollection')
           )
-          .then((collection) => collection?.get('id')),
+          .then((collection) => collection),
       [relationship]
     ),
     false
   );
-  return rightSideCollection === undefined ? (
+  return rightSideCollection === null || rightSideCollection === undefined ? (
     <Input.Text isReadOnly />
   ) : hasTreeAccess('Taxon', 'read') ? (
     <QueryComboBox
       field={schema.models.CollectingEventAttribute.strictGetRelationship(
         'hostTaxon'
       )}
-      forceCollection={rightSideCollection}
+      forceCollection={rightSideCollection.get('id')}
       formType={formType}
       id={id}
       isRequired={isRequired}
       mode={mode}
       relatedModel={schema.models.Taxon}
       resource={resource}
       typeSearch={hostTaxonTypeSearch}
+      disableCanAdd={
```

Comment:
This is not a good fix.
The query combo box receives `forceCollection` prop, which it can use to figure out if current collection is equal to `forceCollection`

In fact, this fix is already implemented. See https://github.com/specify/specify7/pull/3283

In there, I added support for properly creating resource into a correct discipline: https://github.com/specify/specify7/pull/3283/files#diff-11908f1e0fbcca3761cf5d0d0635a57bbedbb34f38675283d64fa843a9af23f0R263-R268

But also, it disables the add feature (temporary). It's disabled only because tree rank pick list is loading data from a wrong discipline:
https://github.com/specify/specify7/pull/3283/files#diff-11908f1e0fbcca3761cf5d0d0635a57bbedbb34f38675283d64fa843a9af23f0R359-R364

Can you please review the code change on that pull request, and undo the query combo box changes on this branch